model/resource/service_experimental.yaml

```diff
@@ -31,7 +31,36 @@ groups:
           the ID is ephemeral and changes during important lifetime events for the service
           (e.g. service restarts).
           If the service has no inherent unique ID that can be used as the value of this attribute
-          it is recommended to generate a random Version 1 or Version 4 RFC 4122 UUID
-          (services aiming for reproducible UUIDs may also use Version 5, see RFC 4122
-          for more recommendations).
+          it is recommended to generate a random Version 1 or Version 4
+          [RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. Services aiming for reproducible UUIDs may also
+          use Version 5. UUIDs are typically recommended, as we only need an opaque yet reproducible value for
+          the purposes of identifying a service instance. Similar to what can be seen in the man page for the
+          `/etc/machine-id` file, the underlying data, such as pod name and namespace should be treated as
+          confidential by this algorithm, being the user's choice to expose it or not via another resource attribute. 
+
+          SDKs are required to follow the following algorithm when generating `service.instance.id`:
+
+          - If the user has provided a `service.instance.id`, via environment
+            variable, configuration or custom resource detection, this will
+            always be used and honored over generated IDs.
+          - When any of the below combinations of resource attribute are provided, they should be used as the input
+            for generating a UUID v5. The values within each combination should be separated with dots:
+            * `container.id`
+            * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`
+            * `host.id`
```

Comment:
Should the service.name be required when host.id is used? Otherwise, multiple applications on the same host would yield the same UUID. If so, should all of the above also include the service.name? It should be safe, as service.name is required anyway.